lib/elixir/lib/kernel.ex

```diff
@@ -1354,9 +1354,12 @@ defmodule Kernel do
       [1, 2 | 3]
 
   """
-  @spec list ++ term :: maybe_improper_list
-  def left ++ right do
-    :erlang.++(left, right)
+  @spec list ++ list :: list
+  defmacro left ++ right do
```

Comment:
Since this function is inlining by the compiler, it should not be a macro. The changes on elixir_utils.erl should be enough, can you please revert this? Thanks!

Reply:
Looks like you already did it! Thanks! :)